stetho/src/main/java/com/facebook/stetho/inspector/elements/ApplicationNodeDescriptor.java

```diff
@@ -0,0 +1,161 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+package com.facebook.stetho.inspector.elements;
+
+import android.app.Activity;
+import android.app.Application;
+import android.os.Bundle;
+
+import com.facebook.stetho.common.Util;
+import com.facebook.stetho.inspector.protocol.module.DOM;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+final class ApplicationNodeDescriptor extends ChainedNodeDescriptor<Application> {
+  private ConcurrentMap<Application, ElementContext> mElementToContextMap =
+      new ConcurrentHashMap<>();
```

Comment:
Never use this on Android.  It is designed for high throughput in highly concurrent environments at the expensive of RAM.  Use `Collections.synchronizedMap(new HashMap...` instead.

As an aside, you don't need a map.  `Application` is a singleton in Android.